bpf mock tests print lots of warnings but succeed

When building bpf mock tests locally, they complain a lot but the build succeeds. It would be nice to quieten these as they are noisy and can mislead developers into thinking there's a problem here when the build actually succeeds:

```
$ make -C test/bpf/ nat-test
```
...
```
Creating mock for conntrack_stub...
In file included from nat-test.c:8:
In file included from ../../bpf/tests/nat_test.h:25:
In file included from ../../bpf/mock/mocks/mock_helpers.h:6:
In file included from ../../bpf/mock/mocks/helpers.h:10:
In file included from ../../bpf/include/bpf/compiler.h:8:
../../bpf/include/linux/../bpf/stddef.h:7:9: warning: 'NULL' macro redefined [-Wmacro-redefined]
#define NULL    ((void *)0)
        ^
/usr/lib/llvm-10/lib/clang/10.0.0/include/stddef.h:89:11: note: previous definition is here
#  define NULL ((void*)0)
          ^
In file included from nat-test.c:8:
In file included from ../../bpf/tests/nat_test.h:28:
../../bpf/mock/fake_maps.h:55:10: warning: incompatible pointer to integer conversion returning 'typeof ((map)->map_types->t_data)' (aka 'void *') from a function with result type 'int' [-Wint-conversion]
  return hashmap_remove(map, key);
         ^~~~~~~~~~~~~~~~~~~~~~~~
../../../hashmap/include/hashmap.h:221:32: note: expanded from macro 'hashmap_remove'
#define hashmap_remove(h, key) ({                                       \
                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from nat-test.c:8:
../../bpf/tests/nat_test.h:128:27: warning: incompatible pointer types passing 'struct (anonymous struct at ../../bpf/tests/nat_test.h:106:1) *' to parameter of type 'hashmap_void_t *' [-Wincompatible-pointer-types]
  return fake_lookup_elem(&ipv4_ct_tuple_map, key);
                          ^~~~~~~~~~~~~~~~~~
../../bpf/mock/fake_maps.h:24:40: note: passing argument to parameter 'map' here
void *fake_lookup_elem(hashmap_void_t *map, void *key) {
                                       ^
In file included from nat-test.c:8:
../../bpf/tests/nat_test.h:128:47: warning: passing 'const void *' to parameter of type 'void *' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
  return fake_lookup_elem(&ipv4_ct_tuple_map, key);
                                              ^~~
../../bpf/mock/fake_maps.h:24:51: note: passing argument to parameter 'key' here
void *fake_lookup_elem(hashmap_void_t *map, void *key) {
                                                  ^
In file included from nat-test.c:8:
../../bpf/tests/nat_test.h:132:27: warning: incompatible pointer types passing 'struct (anonymous struct at ../../bpf/tests/nat_test.h:106:1) *' to parameter of type 'hashmap_void_t *' [-Wincompatible-pointer-types]
  return fake_update_elem(&ipv4_ct_tuple_map, key, value, flags, SNAT_MAPPING_IPV4_SIZE);
                          ^~~~~~~~~~~~~~~~~~
../../bpf/mock/fake_maps.h:29:38: note: passing argument to parameter 'map' here
int fake_update_elem(hashmap_void_t *map, void *key, void *value, __u32 flags, size_t size) {
                                     ^
In file included from nat-test.c:8:
../../bpf/tests/nat_test.h:132:47: warning: passing 'const void *' to parameter of type 'void *' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
  return fake_update_elem(&ipv4_ct_tuple_map, key, value, flags, SNAT_MAPPING_IPV4_SIZE);
                                              ^~~
../../bpf/mock/fake_maps.h:29:49: note: passing argument to parameter 'key' here
int fake_update_elem(hashmap_void_t *map, void *key, void *value, __u32 flags, size_t size) {
                                                ^
In file included from nat-test.c:8:
../../bpf/tests/nat_test.h:132:52: warning: passing 'const void *' to parameter of type 'void *' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
  return fake_update_elem(&ipv4_ct_tuple_map, key, value, flags, SNAT_MAPPING_IPV4_SIZE);
                                                   ^~~~~
../../bpf/mock/fake_maps.h:29:60: note: passing argument to parameter 'value' here
int fake_update_elem(hashmap_void_t *map, void *key, void *value, __u32 flags, size_t size) {
                                                           ^
In file included from nat-test.c:8:
../../bpf/tests/nat_test.h:136:27: warning: incompatible pointer types passing 'struct (anonymous struct at ../../bpf/tests/nat_test.h:106:1) *' to parameter of type 'hashmap_void_t *' [-Wincompatible-pointer-types]
  return fake_delete_elem(&ipv4_ct_tuple_map, key);
                          ^~~~~~~~~~~~~~~~~~
../../bpf/mock/fake_maps.h:54:38: note: passing argument to parameter 'map' here
int fake_delete_elem(hashmap_void_t *map, void *key) {
                                     ^
In file included from nat-test.c:8:
../../bpf/tests/nat_test.h:136:47: warning: passing 'const void *' to parameter of type 'void *' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
  return fake_delete_elem(&ipv4_ct_tuple_map, key);
                                              ^~~
../../bpf/mock/fake_maps.h:54:49: note: passing argument to parameter 'key' here
int fake_delete_elem(hashmap_void_t *map, void *key) {
                                                ^
In file included from nat-test.c:8:
../../bpf/tests/nat_test.h:146:19: warning: incompatible pointer types passing 'struct (anonymous struct at ../../bpf/tests/nat_test.h:106:1) *' to parameter of type 'hashmap_void_t *' [-Wincompatible-pointer-types]
    fake_init_map(&ipv4_ct_tuple_map, bpf_hash_ipv4_ct_tuple, bpf_compare_ipv4_ct_tuple);
                  ^~~~~~~~~~~~~~~~~~
../../bpf/mock/fake_maps.h:19:36: note: passing argument to parameter 'map' here
void fake_init_map(hashmap_void_t *map, size_t (*hash)(const void *key), int (*compare)(const void *a, const void *b)) {
                                   ^
10 warnings generated.
In file included from ../../bpf/mock/mocks/mock_helpers.c:6:
In file included from ../../bpf/mock/mocks/mock_helpers.h:6:
In file included from ../../bpf/mock/mocks/helpers.h:10:
In file included from ../../bpf/include/bpf/compiler.h:8:
../../bpf/include/linux/../bpf/stddef.h:7:9: warning: 'NULL' macro redefined [-Wmacro-redefined]
#define NULL    ((void *)0)
        ^
/usr/lib/llvm-10/lib/clang/10.0.0/include/stddef.h:89:11: note: previous definition is here
#  define NULL ((void*)0)
          ^
1 warning generated.
In file included from ../../bpf/mock/mocks/mock_helpers_skb.c:6:
In file included from ../../bpf/mock/mocks/mock_helpers_skb.h:6:
In file included from ../../bpf/mock/mocks/helpers_skb.h:10:
In file included from ../../bpf/include/bpf/compiler.h:8:
../../bpf/include/linux/../bpf/stddef.h:7:9: warning: 'NULL' macro redefined [-Wmacro-redefined]
#define NULL    ((void *)0)
        ^
/usr/lib/llvm-10/lib/clang/10.0.0/include/stddef.h:89:11: note: previous definition is here
#  define NULL ((void*)0)
          ^
1 warning generated.
In file included from ../../bpf/mock/mocks/mock_conntrack_stub.c:6:
In file included from ../../bpf/mock/mocks/mock_conntrack_stub.h:6:
In file included from ../../bpf/mock/conntrack_stub.h:15:
In file included from ../../bpf/include/bpf/ctx/skb.h:10:
In file included from ../../bpf/include/bpf/ctx/common.h:10:
In file included from ../../bpf/include/bpf/ctx/../compiler.h:8:
../../bpf/include/linux/../bpf/stddef.h:7:9: warning: 'NULL' macro redefined [-Wmacro-redefined]
#define NULL    ((void *)0)
        ^
/usr/lib/llvm-10/lib/clang/10.0.0/include/stddef.h:89:11: note: previous definition is here
#  define NULL ((void*)0)
          ^
1 warning generated.
```

--- a/bpf/include/bpf/stddef.h
+++ b/bpf/include/bpf/stddef.h
@@ -1,10 +1,13 @@
 /* SPDX-License-Identifier: GPL-2.0 */
 /* Copyright (C) 2020 Authors of Cilium */
 
+#ifndef NULL
+#define NULL ((void *)0)
+#endif
+
 #ifndef __BPF_STDDEF_H_
 #define __BPF_STDDEF_H_
 
-#define NULL	((void *)0)
 
 #define bool	_Bool
 

--- a/bpf/mock/fake_maps.h
+++ b/bpf/mock/fake_maps.h
@@ -21,23 +21,26 @@ void fake_init_map(hashmap_void_t *map, size_t (*hash)(const void *key), int (*c
 }
 
 // Loop up the value of given key.
-void *fake_lookup_elem(hashmap_void_t *map, void *key) {
+void *fake_lookup_elem(hashmap_void_t *map, const void *key)
+{
   return hashmap_get(map, key);
 }
 
 // Update the value of given key.
-int fake_update_elem(hashmap_void_t *map, void *key, void *value, __u32 flags, size_t size) {
+int fake_update_elem(hashmap_void_t *map, const void *key, const void *value, __u32 flags,
+		     size_t size)
+{
   void *entry = hashmap_get(map, key);
   if (entry == NULL && hashmap_size(map) >= size) {
       printf("Update failed: Map is full\n");
       return -1;
     }
 
-  int r = hashmap_put(map, key, value);
+  int r = hashmap_put(map, key, (void *)value);
   if (r == -EEXIST) {
     if (flags == BPF_NOEXIST) return -1;
     hashmap_remove(map, key);
-    hashmap_put(map, key, value);
+    hashmap_put(map, key, (void *)value);
     return 0;
   } else if (r == 0) {
     if (flags == BPF_EXIST) {
@@ -51,6 +54,7 @@ int fake_update_elem(hashmap_void_t *map, void *key, void *value, __u32 flags, s
 }
 
 // Delete given key.
-int fake_delete_elem(hashmap_void_t *map, void *key) {
-  return hashmap_remove(map, key);
+int fake_delete_elem(hashmap_void_t *map, const void *key)
+{
+  return (int)hashmap_remove(map, key);
 }

--- a/bpf/tests/nat_test.h
+++ b/bpf/tests/nat_test.h
@@ -125,15 +125,16 @@ size_t bpf_hash_ipv4_ct_tuple(const void *key)
 // encoded in the callback functions as required by the stub functions in CMock.
 // So we need different callback funcs for different fake maps.
 void *ipv4_ct_tuple_map_lookup_elem_callback(const void* map, const void* key, int cmock_num_calls) {
-  return fake_lookup_elem(&ipv4_ct_tuple_map, key);
+  return fake_lookup_elem((hashmap_void_t *)&ipv4_ct_tuple_map, key);
 }
 
 int ipv4_ct_tuple_map_update_elem_callback(const void* map, const void* key, const void* value, __u32 flags, int cmock_num_calls) {
-  return fake_update_elem(&ipv4_ct_tuple_map, key, value, flags, SNAT_MAPPING_IPV4_SIZE);
+  return fake_update_elem((hashmap_void_t *)&ipv4_ct_tuple_map, key, value, flags,
+			 SNAT_MAPPING_IPV4_SIZE);
 }
 
 int ipv4_ct_tuple_map_delete_elem_callback(const void* map, const void* key, int cmock_num_calls) {
-  return fake_delete_elem(&ipv4_ct_tuple_map, key);
+  return fake_delete_elem((hashmap_void_t *)&ipv4_ct_tuple_map, key);
 }
 
 void test_snat_v4_new_mapping() {
@@ -143,7 +144,8 @@ void test_snat_v4_new_mapping() {
     struct ipv4_nat_target target;
 
     // Initiate the map.
-    fake_init_map(&ipv4_ct_tuple_map, bpf_hash_ipv4_ct_tuple, bpf_compare_ipv4_ct_tuple);
+    fake_init_map((hashmap_void_t *)&ipv4_ct_tuple_map, bpf_hash_ipv4_ct_tuple,
+		  bpf_compare_ipv4_ct_tuple);
 
     get_prandom_u32_ExpectAndReturn(0);
     // Stub the map helpers with the callbacks defined above.

